packages/apps/dashboard/server/src/modules/details/details.spec.ts

```diff
@@ -0,0 +1,195 @@
+  });
+
+  it('should fetch and return leaders with reputations', async () => {
+    const mockLeaders = [{ address: '0x123', role: 'Reputation Oracle' }];
+    const mockReputations = [{ address: '0x123', reputation: 'hign' }];
+
+    jest
+      .spyOn(OperatorUtils, 'getLeaders')
+      .mockResolvedValue(mockLeaders as ILeader[]);
+    jest
+      .spyOn(httpService as any, 'get')
+      .mockReturnValue(of({ data: mockReputations }));
+
+    const result = await service.getLeaders(ChainId.ALL);
+
+    expect(result).toEqual([
+      expect.objectContaining({
+        address: '0x123',
+        role: 'Reputation Oracle',
+        reputation: 'hign',
+      }),
+    ]);
+  });
+
+  it('should handle missing reputation data gracefully', async () => {
+    const mockLeaders = [{ address: '0x456', role: 'Job Launcher' }];
+
+    jest
+      .spyOn(OperatorUtils, 'getLeaders')
+      .mockResolvedValue(mockLeaders as ILeader[]);
+    jest.spyOn(httpService as any, 'get').mockReturnValue(of({ data: [] }));
+
+    const result = await service.getLeaders(ChainId.ALL);
+
+    expect(result).toEqual([
+      expect.objectContaining({
+        address: '0x456',
+        role: 'Job Launcher',
+        reputation: 'low',
+      }),
+    ]);
+  });
+
+  it('should return sorted leaders by reputation', async () => {
+    const mockLeaders = [
+      { address: '0xA', role: 'Job Launcher' },
+      { address: '0xB', role: 'Exchange Oracle' },
+      { address: '0xC', role: 'Exchange Oracle' },
+      { address: '0xD', role: 'Exchange Oracle' },
+    ];
+    const mockReputations = [
+      { address: '0xB', reputation: 'hign' },
+      { address: '0xC', reputation: 'medium' },
+    ];
+
+    jest
+      .spyOn(OperatorUtils, 'getLeaders')
+      .mockResolvedValue(mockLeaders as ILeader[]);
+    jest
+      .spyOn(httpService as any, 'get')
+      .mockReturnValue(of({ data: mockReputations }));
+
+    const result = await service.getLeaders(ChainId.POLYGON_AMOY, {
+      orderBy: LeadersOrderBy.REPUTATION,
+      orderDirection: OrderDirection.DESC,
+    });
+
+    expect(result[0].address).toBe('0xB');
+    expect(result[1].address).toBe('0xC');
+  });
+
+  it('should return first 3 sorted leaders by reputation', async () => {
+    const mockLeaders = [
+      { address: '0xA', role: 'Job Launcher' },
+      { address: '0xB', role: 'Exchange Oracle' },
+      { address: '0xC', role: 'Exchange Oracle' },
+      { address: '0xD', role: 'Exchange Oracle' },
+      { address: '0xE', role: 'Recording Oracle' },
+    ];
+    const mockReputations = [
+      { address: '0xB', reputation: 'hign' },
+      { address: '0xC', reputation: 'hign' },
+      { address: '0xD', reputation: 'medium' },
+      { address: '0xF', reputation: 'medium' },
+      { address: '0xE', reputation: 'low' },
+    ];
+
+    const getLeadersSpy = jest.spyOn(OperatorUtils, 'getLeaders');
+    getLeadersSpy.mockResolvedValue(mockLeaders as ILeader[]);
```

Comment:
Let's use
```
getLeadersSpy.mockImplementation(async ({ first }) => {
      return mockLeaders.slice(0, first) as ILeader[];
    });
```

because it helps to disclose issue with invalid filter passed